### PR TITLE
[EC-184] Update core modules to use new Storage Account defender version

### PR DESCRIPTION
**Pull request by Krusty93**
### List of changes

Update all TF core modules
Set redis common backup storage account to use new defender version
Enable diagnostic logs on selfcare CDN

### Motivation and context

We can save up to 1k€/mo using new pricing model for some storage accounts

### Type of changes

- [ ] Add new resources
- [X] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [X] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

After merge, import the existing Defender for cloud instance:
`terraform import -var-file="./env/prod/terraform.tfvars" module.redis_common_backup_zrs.azurerm_security_center_storage_defender.this[0] /subscriptions/ec285037-c673-4f58-b594-d7c480da4e8b/resourceGroups/io-p-rg-common/providers/Microsoft.Storage/storageAccounts/iopstredisbackup`

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
module.selfcare_cdn.azurerm_monitor_diagnostic_setting.diagnostic_settings_cdn_profile

diagnostic settings should be disabled (we use azure policy policy)

File: src/core/storage_api.tf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 module "storage_api" {
```
Comment (pasqualedevita) [inline src/core/storage_api.tf:1]: I think we need to enable new configuration `use_legacy_defender_version = false` for `api` and `apireplica`

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — COMMENTED** [review]
for health_check_eviction_time_in_min I remember a bug into azurerm provider.
Now I don't know if it's fixed but it generates a little noise into the plan

File: src/core/storage_api.tf (hunk 1 above)
Comment (Krusty93) [reply src/core/storage_api.tf:1]: It's in another [PR](https://github.com/pagopa/io-infra/pull/882)

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**